File: php/src/GitHub/Models/ValidationError_errors_value.php
<?php

namespace Baywet\Githubtododemo\Github\Models;

use Microsoft\Kiota\Abstractions\Serialization\Parsable;
use Microsoft\Kiota\Abstractions\Serialization\ParseNode;
use Microsoft\Kiota\Abstractions\Serialization\SerializationWriter;

/**
 * Composed type wrapper for classes integer, string
*/
class ValidationError_errors_value implements Parsable 
{
    /**
     * @var int|null $integer Composed type representation for type integer
    */
    private ?int $integer = null;
    
    /**
     * @var string|null $string Composed type representation for type string
    */
    private ?string $string = null;
    
    /**
     * Creates a new instance of the appropriate class based on discriminator value
     * @param ParseNode $parseNode The parse node to use to read the discriminator value and create the object
     * @return ValidationError_errors_value
    */
    public static function createFromDiscriminatorValue(ParseNode $parseNode): ValidationError_errors_value {
        $result = new ValidationError_errors_value();
        $mappingValueNode = $parseNode->getChildNode("");
        if ($mappingValueNode !== null) {
            $mappingValue = $mappingValueNode->getStringValue();
        }
        if ($parseNode->getIntegerValue() !== null) {
            $result->setInteger($parseNode->getIntegerValue());
        } else if ($parseNode->getStringValue() !== null) {
            $result->setString($parseNode->getStringValue());
        }
        return $result;
    }

    /**
     * The deserialization information for the current model
     * @return array<string, callable(ParseNode): void>
    */
    public function getFieldDeserializers(): array {
        return [];
    }

    /**
     * Gets the integer property value. Composed type representation for type integer
     * @return int|null
    */
    public function getInteger(): ?int {
        return $this->integer;
    }

    /**
     * Gets the string property value. Composed type representation for type string
     * @return string|null
    */
    public function getString(): ?string {
        return $this->string;
    }

    /**
     * Serializes information the current object
     * @param SerializationWriter $writer Serialization writer to use to serialize this model
    */
    public function serialize(SerializationWriter $writer): void {
        if ($this->getInteger() !== null) {
            $writer->writeIntegerValue(null, $this->getInteger());
        } else if ($this->getString() !== null) {
            $writer->writeStringValue(null, $this->getString());
        }
    }

    /**
     * Sets the integer property value. Composed type representation for type integer
     * @param int|null $value Value to set for the integer property.
    */
    public function setInteger(?int $value): void {
        $this->integer = $value;
    }

    /**
     * Sets the string property value. Composed type representation for type string
     * @param string|null $value Value to set for the string property.
    */
    public function setString(?string $value): void {
        $this->string = $value;
    }

}

File: python/githubclient/models/validation_error_errors_value.py
from __future__ import annotations
from collections.abc import Callable
from dataclasses import dataclass, field
from kiota_abstractions.serialization import ComposedTypeWrapper, Parsable, ParseNode, SerializationWriter
from typing import Any, Optional, TYPE_CHECKING, Union

@dataclass
class ValidationError_errors_value(ComposedTypeWrapper, Parsable):
    """
    Composed type wrapper for classes int, list[str], str
    """
    # Composed type representation for type int
    integer: Optional[int] = None
    # Composed type representation for type list[str]
    string: Optional[list[str]] = None
    # Composed type representation for type str
    validation_error_errors_value_string: Optional[str] = None
    
    @staticmethod
    def create_from_discriminator_value(parse_node: ParseNode) -> ValidationError_errors_value:
        """
        Creates a new instance of the appropriate class based on discriminator value
        param parse_node: The parse node to use to read the discriminator value and create the object
        Returns: ValidationError_errors_value
        """
        if parse_node is None:
            raise TypeError("parse_node cannot be null.")
        try:
            child_node = parse_node.get_child_node("")
            mapping_value = child_node.get_str_value() if child_node else None
        except AttributeError:
            mapping_value = None
        result = ValidationError_errors_value()
        if integer_value := parse_node.get_int_value():
            result.integer = integer_value
        elif validation_error_errors_value_string_value := parse_node.get_str_value():
            result.validation_error_errors_value_string = validation_error_errors_value_string_value
        elif string_value := parse_node.get_collection_of_primitive_values(str):
            result.string = string_value
        return result
    
    def get_field_deserializers(self,) -> dict[str, Callable[[ParseNode], None]]:
        """
        The deserialization information for the current model
        Returns: dict[str, Callable[[ParseNode], None]]
        """
        return {}
    
    def serialize(self,writer: SerializationWriter) -> None:
        """
        Serializes information the current object
        param writer: Serialization writer to use to serialize this model
        Returns: None
        """
        if writer is None:
            raise TypeError("writer cannot be null.")
        if self.integer:
            writer.write_int_value(None, self.integer)
        elif self.validation_error_errors_value_string:
            writer.write_str_value(None, self.validation_error_errors_value_string)
        elif self.string:
            writer.write_collection_of_primitive_values(None, self.string)
    

